engine/src/types.rs

```diff
@@ -219,6 +335,98 @@ impl<'a> LhsValue<'a> {
             LhsValue::Bytes(bytes) => LhsValue::Bytes(Cow::Borrowed(bytes)),
             LhsValue::Int(integer) => LhsValue::Int(*integer),
             LhsValue::Bool(b) => LhsValue::Bool(*b),
+            LhsValue::Map(m) => LhsValue::Map(m.clone()),
+        }
+    }
+
+    /// Retrieve an element from an LhsValue given a path item and a specified
+    /// type.
+    /// Returns a TypeMismatchError error if current type does not support it
+    /// nested element. Only LhsValue::Map supports nested elements for now.
+    pub fn get(
```

Comment:
I'd not expose these methods at top level - if paths make sense only for maps, then let them be scoped only to them.